src/components/PaginationFooter/Dropdowns/SelectPage.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import type { SelectChangeEvent } from '@mui/material/Select';
+import PageOptions from './PageOptions';
+import DropDown from './DropDown';
+import { usePaginationContext } from '../context/PaginationProvider';
+
+const SelectPage = () => {
+  const { pageInfo, width, totalPages, handleChangePage } = usePaginationContext();
+  const { page } = pageInfo;
+
+  if (width <= 700) {
```

Comment:
now I'm thinking, since you are doing this for all sub components of a pagination, what if we create a simple  hook that returns flags to show each of these components. 

all flags returned from that hook, will prefixed with something like `shouldShowSelectPage`, `ShouldShowButtons`, `ShouldShowX, Y, Z`. With that, we would pass the responsibility of showing/not-showing pagination elements into that hook

Reply:
`shouldShowSelectPage` would for example would still have to be "consumed" somewhere. Either in this component or the component that is responsible for rendering.

But what are the benefits of that? It would add more possible states to the component that is responsible for rendering (thus add complexity).

Also not sure why it would be a hook? A simple util function for each component seems like enough since the output is either `true` or `false`. So there should be no need to "cache" the calculation or its output.

I do think I should avoid using a magic number.

Reply:
I'm with Christofer on this one. Big fan of the component itself handling if it should be rendered or not, giving the parent an overview of which components that are included, but keeps the jsx really clean